### PR TITLE
Call sample directly

### PR DESCRIPTION
**Pull request by rjweeks**
Move backend call logic here.

### CODE REVIEW

File: tests/views/test_collection_exercise.py [hunk 1]
```diff
@@ -269,8 +277,39 @@ def test_upload_sample(self, mock_request):
             "ingestDateTime": "2017-11-06T14:02:24.203+0000"
         }
 
+        survey_data = {
+            "id": "af6ddd8f-7bd0-4c51-b879-ff4b367461c5"
+        }
+
+        exercise_data = [{
```
Comment (tej99) [inline tests/views/test_collection_exercise.py]: survey_data and exercise_data should probably get loaded in from one of the json files in tests/test_data/ directory like at the top of this file.

**Comment by rjweeks** [comment]
[Validate sample before loading to ensure no duplicate sampleunitref values exist](https://trello.com/c/6dXMo7mN/224-validate-sample-before-loading-to-ensure-no-duplicate-sampleunitref-values-exist)

File: tests/views/test_collection_exercise.py (hunk 1 above)
Comment (andrewmil) [reply tests/views/test_collection_exercise.py]: :+1: anything thats large like this should be in it's own file
We can probably clean up how we pull in our test data across the python repos

File: response_operations_ui/controllers/sample_controllers.py [hunk 2]
```diff
@@ -1,19 +1,61 @@
 import logging
 
 import requests
+
```
Comment (andrewmil) [inline response_operations_ui/controllers/sample_controllers.py]: Don't need these blank lines 🎨

File: response_operations_ui/common/filters.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+def get_collection_exercise_by_period(exercises, period):
+    for exercise in exercises:
+        if exercise['exerciseRef'] == period:
+            return exercise
+
+
+def get_case_group_status_by_collection_exercise(case_groups, collection_exercise_id):
+    for case_group in case_groups:
```
Comment (andrewmil) [inline response_operations_ui/common/filters.py]: This code isn't covered in the tests, is it required at all?

File: response_operations_ui/controllers/sample_controllers.py [hunk 4]
```diff
@@ -1,19 +1,61 @@
...
+    if not exercise:
+        return make_response(jsonify({'message': 'Collection exercise not found'}), 404)
+
+    survey_type = 'B'
+    url = f'{app.config["SAMPLE_URL"]}/samples/{survey_type}/fileupload'
+    response = requests.post(url=url, auth=app.config['SAMPLE_AUTH'], files={'file': file})
+
+    try:
+        response.raise_for_status()
+    except (HTTPError, RequestException):
```
Comment (andrewmil) [inline response_operations_ui/controllers/sample_controllers.py]: This exception case isn't tested

File: response_operations_ui/controllers/collection_exercise_controllers.py [hunk 5]
```diff
@@ -164,3 +164,27 @@ def get_case_group_status_by_collection_exercise(case_groups, collection_exercis
...
+    logger.debug('Linking sample summary to collection exercise',
+                 collection_exercise_id=collection_exercise_id,
+                 sample_summary_id=sample_summary_id)
+    url = f'{app.config["COLLECTION_EXERCISE_URL"]}/collectionexercises/link/{collection_exercise_id}'
+    # Currently we only need to link a single sample to a single collection exercise
+    payload = {'sampleSummaryIds': [str(sample_summary_id)]}
+    response = requests.put(url, auth=app.config['COLLECTION_EXERCISE_AUTH'], json=payload)
+
+    if response.status_code == 404:
+        logger.error('Error retrieving collection exercise', collection_exercise_id=collection_exercise_id)
```
Comment (andrewmil) [inline response_operations_ui/controllers/collection_exercise_controllers.py]: These error cases aren't tested

File: response_operations_ui/controllers/sample_controllers.py [hunk 6]
```diff
@@ -1,6 +1,10 @@
 import logging
 
 import requests
+
+from flask import jsonify, make_response
+
```
Comment (andrewmil) [inline response_operations_ui/controllers/sample_controllers.py]: Still don't need these blank lines 🎨

Comment (rjweeks) [reply response_operations_ui/controllers/sample_controllers.py]: Imports optimised

File: response_operations_ui/controllers/sample_controllers.py (hunk 4 above)
Comment (rjweeks) [reply response_operations_ui/controllers/sample_controllers.py]: Test added

**Comment by jcox-dev** [comment]
[response-operations-ui should directly call sample service (not via backstage)](https://trello.com/c/3mxeXNP9/98-response-operations-ui-should-directly-call-sample-service-not-via-backstage)

**Pull request merged**